src/Device/AndroidSensors.cpp

```diff
@@ -287,6 +288,36 @@ DeviceDescriptor::OnHeartRateSensor(unsigned bpm) noexcept
   e.Commit();
 }
 
+void
+DeviceDescriptor::OnEngineSensors(Temperature cht,
+                                  Temperature egt,
+                                  Temperature oat,
+                                  AtmosphericPressure pressure,
+                                  uint16_t revs_per_sec) noexcept
+{
+  const auto e = BeginEdit();
+  NMEAInfo &basic = *e;
+  basic.UpdateClock();
+  basic.alive.Update(basic.clock);
+  basic.engine_state.revs_per_sec = revs_per_sec;
+  basic.engine_state.revs_per_sec_available.Update(basic.clock);
+
+  if(pressure.IsPlausible()){
+    basic.static_pressure = pressure;
+    basic.static_pressure_available.Update(basic.clock);
+  }else{
+    basic.static_pressure_available.Clear();
+  }
```

Comment:
The same as for `oat` applies here - there's already `OnBarometricPressureSensor()`, making this parameter redundant.

And now you see one specific advantage of not duplicating code: the existing `OnBarometricPressureSensor()` implementation derives the vario value from pressure changes using a Kalman filter - a feature that is missing in your PR. You duplicated code, but your duplication is incomplete.